pilot/cmd/pilot-agent/status/server_test.go

```diff
@@ -186,6 +183,27 @@ my_metric{} 0
 my_other_metric{} 0
 `,
 		},
+		// When the application and envoy share a metric, Prometheus will fail. This negative check validates this
+		// assumption.
+		{
+			name: "conflict metric",
+			envoy: `# TYPE my_metric counter
+my_metric{} 0
+# TYPE my_other_metric counter
+my_other_metric{} 0
+`,
+			app: `# TYPE my_metric counter
+my_metric{} 0
+`,
+			output: `# TYPE my_metric counter
+my_metric{} 0
+# TYPE my_other_metric counter
+my_other_metric{} 0
+# TYPE my_metric counter
+my_metric{} 0
```

Comment:
do we have a test where `my_metric_counter` has labels in one serialization?

Reply:
No I can add it though. The parsing actually fails on the TYPE part before it gets to the metric so I assumed it would fail, but worth adding

Reply:
If that is where the failure is, does that imply a path forward for the related PR?

Reply:
No I think it makes it even worse, we cannot have two duplicate metrics even if they have unique labels? So we need to prefix it.

To make sure I am using prefix/label correctly, I mean if we have existing metric

`go_gc_seconds{}`

We need to make `istio_agent_go_gc_seconds{}`; `go_gc_seconds{type="istio-agent"}` will not work.

Reply:
by directly prefixing? as in prefix+metric? or namespacing?

Reply:
I added the label, same error:
```
    TestStats/conflict_metric: server_test.go:263: got expected error: text format parsing error in line 5: second TYPE line for metric name "my_metric", or TYPE reported after samples
=== RUN   TestStats/conflict_metric_labeled
    TestStats/conflict_metric_labeled: server_test.go:263: got expected error: text format parsing error in line 3: second TYPE line for metric name "my_metric", or TYPE reported after samples
--- PASS: TestStats (0.01s)
```